src/main/java/com/zufar/onlinestore/payment/model/Payment.java

```diff
@@ -15,38 +24,48 @@
 public class Payment {
 
     @Id
-    @GeneratedValue(strategy = GenerationType.UUID)
-    private String paymentId;
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "payment_id")
+    private Long paymentId;
 
-    @Column(nullable = false)
+    @Column(nullable = false, name = "currency")
     private String currency;
 
-    @Column(nullable = false)
-    private BigDecimal totalPrice;
+    @Column(nullable = false, name = "items_total_price")
+    private BigDecimal itemsTotalPrice;
 
+    @Column(name = "status")
     private String status;
 
+    @Column(name = "description")
     private String description;
 
-
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (!(o instanceof Payment payment)) return false;
-        return Objects.equals(paymentId, payment.paymentId) && Objects.equals(currency, payment.currency) && Objects.equals(totalPrice, payment.totalPrice) && status == payment.status && Objects.equals(description, payment.description);
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof Payment)) {
+            return false;
+        }
+        Payment that = (Payment) obj;
+        EqualsBuilder eb = new EqualsBuilder();
+        eb.append(paymentId, that.paymentId);
+        return eb.isEquals();
     }
 
     public int hashCode() {
-        return Objects.hash(paymentId, currency, totalPrice, status, description);
+        HashCodeBuilder hcb = new HashCodeBuilder();
+        hcb.append(paymentId);
+        return hcb.toHashCode();
```

Comment:
If paymentId is auto generated then "hcb.toHashCode()" should not be used.
Use getClass().hashCode() instead

Reply:
equals/hashCode is needed to use inside hash-based collections.
Hence, If we don`t want to use them inside hash-based collections we don`t need to implement them.